plugin/core/compat.gradle

```diff
@@ -0,0 +1,56 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
+String gw = Os.isFamily(Os.FAMILY_WINDOWS) ? 'gradlew.bat' : './gradlew'
+String compatClassesDestination = 'build/compat/classes'
+TaskContainer taskContainer = project.tasks
+
+File[] compatModulesDirs = rootProject.file('plugin/compat').listFiles(new FilenameFilter() {
+
+    /**
+     * We want to collect all the folders inside plugin/compat that are also standalone Gradle projects,
+     * hence we look for their settings.gradle
+     */
+    @Override
+    boolean accept(File dir, String name) {
+        return new File(dir, "$name/settings.gradle").exists()
+    }
+})
+
+Task compatClassesTask = taskContainer.create(name: 'releasePluginCompatClasses', type: DefaultTask) {
+    outputs.dir(compatClassesDestination)
+}
+
+compatModulesDirs.each { File moduleDir ->
+
+    def moduleName = moduleDir.name
+
+    def cleanModuleTask = taskContainer.create(name: "clean${moduleName.capitalize()}", type: Delete) {
```

Comment:
we add a custom clean task to ensure that we clean all the outputs of each compat module when we invoke `:clean` on the `plugin/core` module

Reply:
Shouldn't each module already have a `clean` task we could use instead of creating a new one?

Reply:
they do, but they are also different Gradle projects (see description). The only way to invoke their `:clean` task is then to run a gradle build from the core module Gradle build, but it sounded overkill considering that we need to just remove a folder